### PR TITLE
Fix UTD when hiding member state events.

### LINKED ISSUE #2486: UTD for events before invitation if member state events are hidden
To reproduce:
- User A creates an empty room
- User A sends some messages
- User A invites User B
- User B joins the room
- User B switches off the "Show room member state events" settings from Settings>Preferences
- User B navigates to the room timeline and see UTD for the previous events which User A sent

### PR DESCRIPTION
**Pull request by onurays**
Fixes #2486

### CODE REVIEW

**Review by bmarty — APPROVED** [review]
LGTM, thanks!

**Comment by bmarty** [comment]
@onurays could you fix the copyright issue reported by the CI please? Thanks

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/DefaultTimeline.kt [hunk 1]
```diff
@@ -785,7 +785,7 @@ internal class DefaultTimeline(
 
     private fun List<TimelineEvent>.filterEventsWithSettings(): List<TimelineEvent> {
         return filter {
-            val filterType = !settings.filters.filterTypes || settings.filters.allowedTypes.contains(it.root.type)
+            val filterType = !settings.filters.filterTypes || settings.filters.allowedTypes.map { it.eventType }.contains(it.root.type)
```
Comment (ganfra) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/DefaultTimeline.kt:788]: You should apply the same logic than in TimelineEventEntityQueries.kt

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/TimelineHiddenReadReceipts.kt [hunk 2]
```diff
@@ -151,7 +151,7 @@ internal class TimelineHiddenReadReceipts constructor(private val readReceiptsSu
         beginGroup()
         var needOr = false
         if (settings.filters.filterTypes) {
-            val allowedTypes = settings.filters.allowedTypes.toTypedArray()
+            val allowedTypes = settings.filters.allowedTypes.map { it.eventType }.toTypedArray()
```
Comment (ganfra) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/TimelineHiddenReadReceipts.kt:154]: Same remark ^

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/TimelineSettingsFactory.kt [hunk 3]
```diff
@@ -48,12 +53,12 @@ class TimelineSettingsFactory @Inject constructor(private val userPreferencesPro
         }
     }
 
-    private fun List<String>.filterDisplayableTypes(): List<String> {
-        return filter { type ->
-            when (type) {
-                EventType.STATE_ROOM_MEMBER -> userPreferencesProvider.shouldShowRoomMemberStateEvents()
-                else                        -> true
-            }
+    private fun List<String>.filterDisplayableTypes(): List<EventTypeFilter> {
```
Comment (ganfra) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/TimelineSettingsFactory.kt:56]: Maybe change the name of the method to reflect mapping?

**Review by ganfra — CHANGES_REQUESTED** [review]
Some remarks

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/DefaultTimeline.kt (hunk 1 above)
Comment (onurays) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/DefaultTimeline.kt]: Done.

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/TimelineHiddenReadReceipts.kt (hunk 2 above)
Comment (onurays) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/TimelineHiddenReadReceipts.kt]: Done.

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/TimelineSettingsFactory.kt (hunk 3 above)
Comment (onurays) [reply vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/TimelineSettingsFactory.kt]: Done.

**Review by ganfra — APPROVED** [review]
Didn't tested but LGTM now, thanks

**Pull request merged**